src/Authoring/WinRT.SourceGenerator/RcwReflectionFallbackGenerator.cs

```diff
@@ -0,0 +1,332 @@
+
+            context.AddSource("RcwFallbackInitializer.g.cs", builder.ToString());
+        });
+    }
+
+    /// <summary>
+    /// Checks whether an assembly contains old projections.
+    /// </summary>
+    /// <param name="assemblySymbol">The assembly to inspect.</param>
+    /// <returns>Whether <paramref name="assemblySymbol"/> contains old projections.</returns>
+    private static bool IsOldProjectionAssembly(IAssemblySymbol assemblySymbol)
+    {
+        // We only care about assemblies that have some dependent assemblies
+        if (assemblySymbol.Modules.First() is not { ReferencedAssemblies: { Length: > 0 } dependentAssemblies })
+        {
+            return false;
+        }
+
+        // Scan all dependent assemblies to look for CsWinRT with version < 2.0.8
+        foreach (AssemblyIdentity assemblyIdentity in dependentAssemblies)
+        {
+            if (assemblyIdentity.Name == "WinRT.Runtime")
+            {
+                return assemblyIdentity.Version < new Version(2, 0, 8);
+            }
+        }
+
+        // This assembly is not a projection assembly
+        return false;
+    }
+
+    /// <summary>
+    /// Visits all named type symbols in a given assembly, except for ABI types.
+    /// </summary>
+    /// <param name="assemblySymbol">The assembly to inspect.</param>
+    /// <returns>All named type symbols in <paramref name="assemblySymbol"/>, except for ABI types.</returns>
+    private static IEnumerable<INamedTypeSymbol> VisitNamedTypeSymbolsExceptABI(IAssemblySymbol assemblySymbol)
+    {
+        static IEnumerable<INamedTypeSymbol> Visit(INamespaceOrTypeSymbol symbol)
+        {
+            foreach (ISymbol memberSymbol in symbol.GetMembers())
+            {
+                // Visit the current symbol if it's a type symbol
+                if (memberSymbol is INamedTypeSymbol typeSymbol)
+                {
+                    yield return typeSymbol;
+                }
+                else if (memberSymbol is INamespaceSymbol { Name: not ("ABI" or "WinRT") } namespaceSymbol)
+                {
+                    // If the symbol is a namespace, also recurse (ignore the ABI namespaces)
+                    foreach (INamedTypeSymbol nestedTypeSymbol in Visit(namespaceSymbol))
+                    {
+                        yield return nestedTypeSymbol;
+                    }
+                }
+            }
+        }
+
+        return Visit(assemblySymbol.GlobalNamespace);
+    }
+
+    /// <summary>
+    /// Checks whether a given type symbol has an attribute with a specified type.
+    /// </summary>
+    /// <param name="typeSymbol">The type to check.</param>
+    /// <param name="attributeTypeSymbol">The attribute to look for.</param>
+    /// <returns>Whether <paramref name="typeSymbol"/> has an attribute with type <paramref name="attributeTypeSymbol"/>.</returns>
+    private static bool ContainsAttributeWithType(ITypeSymbol typeSymbol, ITypeSymbol attributeTypeSymbol)
```

Comment:
Same as Generator.GeneratorHelper.HasAttributeWithType

Reply:
Mmh not seeing any methods with that name 🤔

Reply:
https://github.com/microsoft/CsWinRT/blob/3b983c31ac1889bc78bd45b422bb6b49f0c58ae6/src/Authoring/WinRT.SourceGenerator/Helper.cs#L320

Reply:
Ooh I see, it was added in a recent PR and this branch was outdated ahah, thanks! 😄